zkevm-circuits/src/evm_circuit/execution/calldatasize.rs

```diff
@@ -89,108 +89,98 @@ impl<F: Field> ExecutionGadget<F> for CallDataSizeGadget<F> {
-                call_id,
-                field_tag: CallContextFieldTag::CallDataLength,
-                value: Word::from(call_data_size),
-            }],
-        );
-        rw_map.insert(
-            RwTableTag::Stack,
-            vec![Rw::Stack {
-                rw_counter: 10,
-                is_write: true,
-                call_id,
-                stack_pointer: 1023,
-                value: Word::from(call_data_size),
-            }],
-        );
 
-        let steps = vec![
-            ExecStep {
-                execution_state: ExecutionState::CALLDATASIZE,
-                rw_indices: vec![(RwTableTag::CallContext, 0), (RwTableTag::Stack, 0)],
-                rw_counter: 9,
-                program_counter: 0,
-                stack_pointer: 1024,
-                gas_left: OpcodeId::CALLDATASIZE.constant_gas_cost().as_u64(),
-                gas_cost: OpcodeId::CALLDATASIZE.constant_gas_cost().as_u64(),
-                opcode: Some(OpcodeId::CALLDATASIZE),
-                ..Default::default()
-            },
-            ExecStep {
-                execution_state: ExecutionState::STOP,
-                rw_counter: 11,
-                program_counter: 1,
-                stack_pointer: 1023,
-                gas_left: 0,
-                opcode: Some(OpcodeId::STOP),
-                ..Default::default()
-            },
-        ];
-
-        let block = Block {
-            randomness,
-            txs: vec![Transaction {
-                id: 1,
-                call_data,
-                call_data_length: call_data_size,
-                steps,
-                calls: vec![Call {
-                    id: call_id,
-                    is_root,
-                    is_create: false,
-                    call_data_length: call_data_size as u64,
-                    code_source: CodeSource::Account(bytecode.hash),
-                    ..Default::default()
-                }],
-                ..Default::default()
-            }],
-            rws: RwMap(rw_map),
-            bytecodes: vec![bytecode],
-            ..Default::default()
+        let block_data = if is_root {
+            bus_mapping::mock::BlockData::new_from_geth_data(
+                TestContext::<2, 1>::new(
+                    None,
+                    |accs| {
+                        accs[0]
+                            .address(address!("0x0000000000000000000000000000000000000000"))
+                            .balance(Word::from(1u64 << 30));
+                        accs[1]
+                            .address(address!("0x0000000000000000000000000000000000000010"))
+                            .balance(Word::from(1u64 << 20))
+                            .code(bytecode);
+                    },
+                    |mut txs, accs| {
+                        txs[0]
+                            .from(accs[0].address)
+                            .to(accs[1].address)
+                            .input(vec![0; call_data_size].into())
```

Comment:
```suggestion
                            .input(rand_bytes(call_data_size).into())
```
I think it's always better to use non-zero values for tests.